lldb/source/Plugins/SyntheticFrameProvider/ScriptedFrameProvider/ScriptedFrameProvider.cpp

```diff
@@ -0,0 +1,232 @@
+                                "Provides synthetic frames via scripting",
+                                ScriptedFrameProvider::CreateInstance);
+}
+
+void ScriptedFrameProvider::Terminate() {
+  PluginManager::UnregisterPlugin(ScriptedFrameProvider::CreateInstance);
+}
+
+llvm::Expected<lldb::SyntheticFrameProviderSP>
+ScriptedFrameProvider::CreateInstance(lldb::ThreadSP thread_sp) {
+  if (!thread_sp)
+    return llvm::createStringError(
+        "failed to create scripted frame: invalid thread");
+
+  ProcessSP process_sp = thread_sp->GetProcess();
+  if (!process_sp)
+    return nullptr;
+
+  Target &target = process_sp->GetTarget();
+
+  Status error;
+  if (auto descriptor = target.GetScriptedFrameProviderDescriptor()) {
+    if (!descriptor->IsValid())
+      return llvm::createStringError(
+          "failed to create scripted frame: invalid scripted metadata");
+
+    if (!descriptor->AppliesToThread(thread_sp->GetID()))
+      return nullptr;
+
+    auto provider_sp = std::make_shared<ScriptedFrameProvider>(
+        thread_sp, *descriptor->scripted_metadata_sp, error);
+    if (!provider_sp || error.Fail())
+      return error.ToError();
+
+    return provider_sp;
+  }
+
+  return llvm::createStringError(
+      "failed to create scripted frame: invalid scripted metadata");
+}
+
+ScriptedFrameProvider::ScriptedFrameProvider(
+    ThreadSP thread_sp, const ScriptedMetadata &scripted_metadata,
+    Status &error)
+    : SyntheticFrameProvider(thread_sp), m_interface_sp(nullptr) {
+  if (!m_thread_sp) {
+    error = Status::FromErrorString(
+        "cannot create scripted frame provider: Invalid thread");
+    return;
+  }
+
+  ProcessSP process_sp = m_thread_sp->GetProcess();
+  if (!process_sp) {
+    error = Status::FromErrorString(
+        "cannot create scripted frame provider: Invalid process");
+    return;
+  }
+
+  ScriptInterpreter *script_interp =
+      process_sp->GetTarget().GetDebugger().GetScriptInterpreter();
+  if (!script_interp) {
+    error = Status::FromErrorString("cannot create scripted frame provider: No "
+                                    "script interpreter installed");
+    return;
+  }
+
+  m_interface_sp = script_interp->CreateScriptedFrameProviderInterface();
+  if (!m_interface_sp) {
+    error = Status::FromErrorString(
+        "cannot create scripted frame provider: Script interpreter couldn't "
+        "create Scripted Frame Provider Interface");
+    return;
+  }
+
+  auto obj_or_err = m_interface_sp->CreatePluginObject(
+      scripted_metadata.GetClassName(), m_thread_sp,
+      scripted_metadata.GetArgsSP());
+  if (!obj_or_err) {
+    error = Status::FromError(obj_or_err.takeError());
+    return;
+  }
+
+  StructuredData::ObjectSP object_sp = *obj_or_err;
+  if (!object_sp || !object_sp->IsValid()) {
+    error = Status::FromErrorString(
+        "cannot create scripted frame provider: Failed to create valid script "
+        "object");
+    return;
+  }
+
+  error.Clear();
+}
```

Comment:
Since this constructor can fail, you need something to say it didn't succeed.  Looks like you are using !m_interface_sp, but if you get an error at this stage, then m_interface_sp has a value, but there was still a construction error?